### PR TITLE
LPS-166700 [Raylife AP] Creation and componentization of the navigation bar and layout

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [LPS-166700](https://issues.liferay.com/browse/LPS-166700)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+type NavigationBar = {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx:15]: Please, change it.
```suggestion
type NavigationBarType = {
```

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	props: NavigationBar
+): React.ReactElement[] => {
+	return props.navbarLabel.map(
+		(currentNavbarLabel: string, index: number) => (
+			<div className="bg-neutral-0 ml-0 mr-0 w-100" key={index}>
+				<ul className="nav nav-pills">
+					<li className="nav-item w-100">
+						<a
+							className={
+								currentNavbarLabel === props.active
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx:31]: Please, use classname library to do this.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+import './index.scss';
+
+import {useState} from 'react';
+
+import NavigationBar from '../../../../../common/components/navigation-bar';
+import {getWebDavUrl} from '../../../../../common/utils/webdav';
+
+const PolicyDetail = () => {
+	const amountOfPoliceDetail = [0];
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:23]: Please, change it.
```suggestion
	const amountOfPolicyDetail = [0];
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+			<div className="d-flex flex-row">
+				<NavigationBar
+					active={active}
+					navbarLabel={navbarLabel}
+					setActive={setActive}
+				/>
+			</div>
+
+			{amountOfPoliceDetail.map(
+				(curentPoliceDetail, indexPoliceDetail) => (
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:42]: Please, put the type of parameters here.
```suggestion
				(curentPoliceDetail:?, indexPoliceDetail:number) => (
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+import './index.scss';
+
+import {useState} from 'react';
+
+import NavigationBar from '../../../../../common/components/navigation-bar';
+import {getWebDavUrl} from '../../../../../common/utils/webdav';
+
+const PolicyDetail = () => {
+	const amountOfPoliceDetail = [0];
+	const navbarLabel = ['Vehicles', 'Drives'];
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx:24]: Please, use an enum for 'Vehicle' and 'Drives'

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx (hunk 1 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx (hunk 2 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/common/components/navigation-bar/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 3 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx (hunk 5 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/PolicyDetail/index.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by silasbsi — APPROVED** [review]

**Comment by silasbsi** [comment]
LGMT :heavy_check_mark:

**Review by joanagrego — APPROVED** [review]
Approved! Thanks @eliasantosf :white_check_mark:

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**